tests/ha/single_pod_restart/test_server_pod_restart.py

```diff
@@ -384,3 +421,300 @@ def test_write_after_server_pod_restart(self):
+                                                    log_prefix=self.test_prefix, skipcleanup=True)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 1: Performed WRITEs/READs-Verify with variable sizes objects.")
+        num_replica = self.num_replica - 1
+        LOGGER.info("Step 2: Shutdown server pod by making replicas=0 and verify cluster & "
+                    "remaining pods status")
+        resp = self.ha_obj.delete_kpod_with_shutdown_methods(
+            master_node_obj=self.node_master_list[0], health_obj=self.hlth_master_list[0],
+            pod_prefix=[const.SERVER_POD_NAME_PREFIX], delete_pod=[self.delete_pod],
+            num_replica=num_replica)
+        # Assert if empty dictionary
+        assert_utils.assert_true(resp[1], "Failed to shutdown/delete server pod")
+        pod_name = list(resp[1].keys())[0]
+        if self.set_type == const.STATEFULSET:
+            self.set_name = resp[1][pod_name]['deployment_name']
+        elif self.set_type == const.REPLICASET:
+            self.deployment_name = resp[1][pod_name]['deployment_name']
+        self.restore_pod = True
+        self.restore_method = resp[1][pod_name]['method']
+        assert_utils.assert_true(resp[0], "Cluster/Services status is not as expected")
+        LOGGER.info("Step 2: Successfully shutdown server pod %s. Verified cluster and "
+                    "services states are as expected & remaining pods status is online.", pod_name)
+        LOGGER.info("Step 3: Perform WRITEs/READs-Verify with variable object sizes and create "
+                    "new bucket in degraded mode")
+        self.test_prefix_deg = 'test-44835-deg'
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                    log_prefix=self.test_prefix_deg,
+                                                    skipcleanup=True,
+                                                    setup_s3bench=False)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 3: Performed WRITEs/READs-Verify with variable object sizes and create "
+                    "new bucket in degraded mode")
+        LOGGER.info("Step 4: Perform READs and verify DI on the data in background of healthy "
+                    "cluster")
+        self.test_prefix = 'test-44835-hlt'
+        args = {'s3userinfo': list(users.values())[0], 'log_prefix': self.test_prefix,
+                'nclients': 1, 'nsamples': 10, 'skipwrite': True, 'skipcleanup': True,
+                'output': output, 'setup_s3bench': False}
+        thread = threading.Thread(target=self.ha_obj.event_s3_operation,
+                                  args=(event,), kwargs=args)
+        thread.daemon = True  # Daemonize thread
+        thread.start()
+        LOGGER.info("Step 4: Successfully started READs and verified DI on the written data in "
```

Comment:
Don't we need to start one more thread for read in background for data written after shutting down server pod?